src/status_im/contacts/db.cljs

```diff
@@ -0,0 +1,89 @@
+
+;;;; DB
+
+;;Contact
+
+(s/def :contact/whisper-identity ::not-empty-string)        ;we can't validate public key, because for dapps whisper-identity is just string
+(s/def :contact/name ::not-empty-string)
+(s/def :contact/address (s/nilable is-address?))            ;nil for Dapps
+(s/def :contact/private-key (s/nilable string?))            ;nil for Dapps
+(s/def :contact/public-key (s/nilable string?))             ;nil for Dapps
+(s/def :contact/photo-path (s/nilable string?))
+(s/def :contact/status (s/nilable string?))
+
+(s/def :contact/last-updated (s/nilable int?))
+(s/def :contact/last-online (s/nilable int?))
+(s/def :contact/pending? boolean?)
+(s/def :contact/unremovable? boolean?)
+
+(s/def :contact/dapp? boolean?)
+(s/def :contact/dapp-url (s/nilable string?))
+(s/def :contact/dapp-hash (s/nilable int?))
+(s/def :contact/bot-url (s/nilable string?))
+(s/def :contact/global-command (s/nilable map?))
+(s/def :contact/commands (s/nilable (s/map-of keyword? map?)))
+(s/def :contact/responses (s/nilable (s/map-of keyword? map?)))
+(s/def :contact/commands-loaded? (s/nilable boolean?))
+(s/def :contact/subscriptions (s/nilable map?))
+(s/def :contact/debug? boolean?)                            ;true when contact added using status-dev-cli
+
+(s/def :contact/contact (allowed-keys
+                          :req-un [:contact/name :contact/whisper-identity]
+                          :opt-un [:contact/address :contact/private-key :contact/public-key :contact/photo-path
+                                   :contact/status :contact/last-updated :contact/last-online :contact/pending?
+                                   :contact/unremovable? :contact/dapp? :contact/dapp-url :contact/dapp-hash
+                                   :contact/bot-url :contact/global-command :contact/commands-loaded?
+                                   :contact/commands :contact/responses :contact/debug? :contact/subscriptions]))
+
+;;Contact list ui props
+(s/def :contact-list-ui/edit? boolean?)
+
+;;Contacts ui props
+(s/def :contacts-ui/edit? boolean?)
+
+
+(s/def :contacts/contacts (s/nilable (s/map-of ::not-empty-string :contact/contact)))
+(s/def :contacts/new-contact-identity (s/nilable string?))  ;public key of new contact during adding this new contact
+(s/def :contacts/new-contact-public-key-error (s/nilable string?))
+(s/def :contacts/contact-identity (s/nilable ::public-key)) ;on showing this contact profile
+(s/def :contacts/contact-list-ui-props (s/nilable (allowed-keys :opt-un [:contact-list-ui/edit?])))
+(s/def :contacts/contacts-ui-props (s/nilable (allowed-keys :opt-un [:contacts-ui/edit?])))
+(s/def :contacts/contacts-click-handler (s/nilable fn?))    ;used in modal list (for example for wallet)
+(s/def :contacts/contacts-click-action (s/nilable #{:send :request})) ;used in modal list (for example for wallet)
+(s/def :contacts/contacts-click-params (s/nilable map?))    ;used in modal list (for example for wallet)
```

Comment:
yenda: I think the first is better because with inline aligned comments you need to manually realign. Also it makes really long lines.
janherich: I also like comment preceding expression little bit more, the main reason is that with inline comments  it’s almost impossible to maintain short overall line length